File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n=nums.length, key=0;
        Arrays.sort(nums);
        int c=1;
        for(int i=0;i<n-1;i++){
            if(nums[i]==nums[i+1]){
                c++;
            }else{
                if(c>n/2){
                    return nums[i];
                }else{
                    c=1;
                }
            }
        }
        return nums[n-1];
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count_map={}
        length=len(nums)
        for n in nums:
            count_map[n]=count_map.get(n,0)+1
        for key, value in count_map.items():
            if value>length//2:
                return key
        return -1